src/installer.ts

```diff
@@ -123,8 +127,70 @@ function normalizeVersion(version: string): string {
     //append minor and patch version if not available
     return version.concat('.0.0');
   } else if (versionPart[2] == null) {
+    // handle beta and rc: 1.10beta1 => 1.10.0-beta1, 1.10rc1 => 1.10.0-rc1
+    if (versionPart[1].includes('beta') || versionPart[1].includes('rc')) {
+      versionPart[1] = versionPart[1]
+        .replace('beta', '.0-beta')
+        .replace('rc', '.0-rc');
+      return versionPart.join('.');
+    }
+
     //append patch version if not available
     return version.concat('.0');
   }
   return version;
 }
+
+async function determineVersion(version: string): Promise<string> {
+  if (!version.endsWith('.x')) {
+    return version;
+  }
+
+  return await getLatestVersion(version);
+}
+
+async function getLatestVersion(version: string): Promise<string> {
+  // clean .x syntax: 1.10.x -> 1.10
+  const trimmedVersion = version.slice(0, version.length - 2);
+
+  const versions = await getPossibleVersions(trimmedVersion);
+
+  core.debug(`evaluating ${versions.length} versions`);
+
+  if (version.length === 0) {
+    core.debug('match not found');
+    return trimmedVersion;
```

Comment:
If no matching versions are found should we just throw? Are there any cases where we don't match a version here, but we succeed later? We might get a clearer error that way